src/Microsoft.VisualStudio.Composition/Configuration/MetadataViewGenerator.cs

```diff
@@ -94,12 +96,34 @@ internal static class MetadataViewGenerator
 
         public delegate object MetadataViewFactory(IReadOnlyDictionary<string, object?> metadata, IReadOnlyDictionary<string, object?> defaultMetadata);
 
-        private static AssemblyBuilder CreateProxyAssemblyBuilder()
+        private static AssemblyBuilder CreateProxyAssemblyBuilder(ImmutableHashSet<AssemblyName> assemblies)
         {
-            var proxyAssemblyName = new AssemblyName(string.Format(CultureInfo.InvariantCulture, "MetadataViewProxies_{0}", Guid.NewGuid()));
+            var proxyAssemblyName = new AssemblyName(string.Format(CultureInfo.InvariantCulture, "MetadataViewProxies_{0}", GenerateGuidFromAssemblies(assemblies)));
             return AssemblyBuilder.DefineDynamicAssembly(proxyAssemblyName, AssemblyBuilderAccess.Run);
         }
 
+        private static Guid GenerateGuidFromAssemblies(ImmutableHashSet<AssemblyName> assemblies)
+        {
+            // To make dll load and JIT comparisons between builds easier, we generate a
+            // consistent name for the proxy assembly based on the input assemblies involved.
+            using var algorithm = SHA256.Create();
+
+            foreach (AssemblyName name in assemblies.OrderBy(a => a.Name, StringComparer.Ordinal))
```

Comment:
Your guid generation must guarantee a unique guid for a given set of runtime assemblies.

I believe this should be `FullName`, because (on .NET Framework) there may be two Utilities assemblies loaded in the same appdomain if their fullname is unique.
Two AppDomains that load the same fullname shouldn't pose a problem because the guid only needs to be unique within one AppDomain.

On .NET, even FullName collisions are possible since one assembly may be loaded in multiple ALCs, but I guess that's a pre-existing bug in the code so no need to worry about that in your PR.

Reply:
Fixed.